### PR TITLE
release-libs action

### PR DESCRIPTION
**Pull request by PietroPasotti**
The action is meant to be used on merge. It will: 
 - fetch the libs owned by the charm from charmhub, compare them with the ones found in libs/charms/charm_name and generate a diff. 
 - print out a list of libs that are going to be published, and/or an error message if there's something wrong.
 - for each bumped lib: publish-lib.

Usage:

```yaml
on:
  push:
    branches:
     - master

jobs:
  release-libs:
    name: Publish any bumped charm libraries
    runs-on: ubuntu-latest
    steps:
      - name: Release libs
        uses: canonical/charming-actions/release-libs@0.1
        with:
          credentials: "${{ secrets.CHARMHUB_TOKEN }}"
          github-token: "${{ secrets.GITHUB_TOKEN }}"
```

### CODE REVIEW

**Review by simskij — COMMENTED** [review]
I know this is still marked as a draft, but as you've said yourself that JS is not your strongest language, I thought I'd leave some early pointers.

File: src/actions/release-libraries/release-libraries.ts [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+  private checkOnly: boolean;
+  private charmName: string;
+  private snap: Snap;
+
+  constructor() {
+    this.tokens = {
+      github: getInput('github-token'),
+      charmhub: getInput('credentials'),
+    };
+    if (!this.tokens.github) throw new Error(`Input 'github-token' is missing`);
```
[the remark this answers is not in the document]
Comment (PietroPasotti) [reply src/actions/release-libraries/release-libraries.ts:28]: noted! In my defense, I ripped that code off of someone else's action.

**Comment by lucabello** [comment]
Looks good to me! I'll wait for this to be tested before approving :)

**Comment by PietroPasotti** [comment]
@lucabello @simskij I verified that it does what it should on three scenarios:
- lib has no changes
- lib has changes but patch is not bumped
- patch is bumped but lib has no changes

what's still to test is: major version bumps

If you want to try it out/see it live: https://github.com/PietroPasotti/test-gh-ci-lib-release/

what's still to be determined is: 
- which of these scenarios should cause the action to fail? For now the logic is: abort and don't publish any changes if there's ANY error at the library collection step. However if some publish-lib fails, the others will still go through.
- does the action have any output (for other actions to use?)

We can however think about these later and take it as it is for now.

**Review by lucabello — APPROVED** [review]
Looks great to me; I'd say that the current behavior and output are good and we don't need anything else for now.

Just a small note: the CI execution reports a `(node:1669) [DEP0018] DeprecationWarning`: if @simskij has a better idea as to where that comes from and how to solve it, that's great; otherwise, for me it can be merged!

**Comment by simskij** [comment]
Pushed the PR content to a feature branch in the main repo. #78 supersedes this PR and includes all changes.

**Pull request closed without merging**